Automattic/wp-calypso
# ci(e2e): Fix test names for TeamCity

#### Changes proposed in this Pull Request

Use `mocha-teamcity-reporter` instead of writing a jUnit file and asking TeamCity to parse it.

This new reporter works by using [TeamCity's Service Messages](https://www.jetbrains.com/help/teamcity/service-messages.html) to report the status of the tests. It should:

* Generates better names without mixing the testsuite and test names.
* Have a more stable test count (the previous junit-based reporter was miscounting some tests every now and then).
* Identify skipped tests (so we can track that metric over time)
* Allow us to mute flaky tests from the UI.

*Before*: Note how it mixes `describe` and `step` names in the test description.
![image](https://user-images.githubusercontent.com/975703/110011730-0d454100-7d20-11eb-9939-89e01205520c.png)

*After*:
![image](https://user-images.githubusercontent.com/975703/110011348-9f007e80-7d1f-11eb-9a40-3664fae3fde8.png)


#### Testing instructions

* Check TeamCity e2e test (build `Run browser e2e tests (desktop)`) they should have a good name.
* Check CircleCI e2e tests, the name should be the same than `trunk`.
